app/src/main/java/org/mozilla/fenix/components/toolbar/CFRPresenter.kt

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.clickable
+import androidx.compose.material.Text
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextDecoration
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.flow.transformWhile
+import mozilla.components.browser.state.selector.findCustomTabOrSelectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.service.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.TrackingProtection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.cfr.CFRPopup
+import org.mozilla.fenix.compose.cfr.CFRPopupProperties
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.settings.SupportUtils
+import org.mozilla.fenix.settings.SupportUtils.SumoTopic.TOTAL_COOKIE_PROTECTION
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Delegate for handling all the business logic for showing CFRs in the toolbar.
+ *
+ * @param context [Context] used for various Android interactions.
+ * @param browserStore [BrowserStore] which will be observed for tabs updates
+ * @param settings [Settings] allowing to read and write persistent user settings
+ * @param toolbar [BrowserToolbar] that will serve as anchor for the CFRs
+ * @param sessionId [String] optional custom tab id used to identify the custom tab in which to show a CFR.
```

Comment:
```suggestion
 * @param sessionId Optional custom tab id [String] used to identify the custom tab in which to show a CFR.
```

Reply:
All parameters are documented with the type at front. "Optional custom tab id [String]" seems forced.

Reply:
The idea was to get the type in the documentation in a standardized way.
| with types | without types |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/11428869/184298396-407ac0bf-7c12-4b99-a8b7-0d09f879c8be.png) | ![image](https://user-images.githubusercontent.com/11428869/184298292-bd5c395e-b231-42b1-853f-5f29ccaea18d.png) |

`@param sessionId [String]` does [already exist in the project](https://searchfox.org/mozilla-mobile/search?q=%40param+sessionId+%5BString%5D&path=&case=false&regexp=false).
`@param Optional custom tab id [String]` or just `tab id [String]` [does not](https://searchfox.org/mozilla-mobile/search?q=tab+id+%5BString%5D&path=&case=false&regexp=false).

Will remove the types.

Reply:
Sorry, I didn't mean to say we had to remove the type from the front or completely. I think the other usages were fine because they were used correctly in a sentence, but the description for sessionId wasn't. The point is that we can still reference the type, but just phrase it correctly grammatically.